### PR TITLE
Stay on same tab when switching domains

### LINKED ISSUE #97446: Retain selected tab when domain is switched
In our new flyout view, when domain is switched, we open the default tab, that is the domain overview tab. So even if we are currently on the Emails tab, if we switch domains, the Overview will be opened for the new domain. Then we'll need to switch to the Emails tab.

Similar to how wordpress.com/sites work, we want to open the same tab when domains are switched

### PR DESCRIPTION
**Pull request by donnapep**
Closes #97446.

TODO:
- [x] Make it work with subpages like Email Forwarding (maybe? Check with Imran)
- [x] Is the Emails tab always guaranteed to be there for all sites? If not, account for that.
- [ ] Add tests as needed
- [ ] Nice to have: Make it scalable for any other tabs that may be added

## Proposed Changes

Add `selectedFeature` prop to ensure UI stays on same tab when switching domains.

## Testing Instructions

* Enable the `calypso/all-domain-management` flag.
* On `/domains/manage`, select a domain.
* Click on the _Emails_ tab.
* Switch to a new domain.
* Ensure the _Emails_ tab is still selected.
* Click on the _Overview_ tab.
* Switch to a new domain.
* Ensure the _Overview_ tab is still selected.
* Disable the `calypso/all-domain-management` flag.
* Click on different domains and ensure the functionality still works as before.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by bogiii** [comment]
> Make it work with subpages like Email Forwarding (maybe? Check with Imran)

We made the decision not to support the subpages.

> Is the Emails tab always guaranteed to be there for all sites? If not, account for that.

I researched the current feature state, and I understand that the Email tab is available for each domain.

Discussion: p1735037561032999/1734725022.860659-slack-C082KU0C0MR

File: packages/domains-table/src/utils/paths.ts [hunk 1]
```diff
@@ -22,7 +23,14 @@ export function domainManagementLink(
...
 
 	if ( isAllDomainManagementEnabled && isAllSitesView ) {
-		return `/domains/manage/all/overview/${ domain }/${ siteSlug }`;
+		switch ( feature ) {
+			case 'email-management':
+				return `/domains/manage/all/email/${ domain }/${ siteSlug }`;
+
+			case 'domain-overview':
+			default:
+				return `/domains/manage/all/overview/${ domain }/${ siteSlug }`;
```
Comment (Imran92) [inline packages/domains-table/src/utils/paths.ts]: Can we use the `domainManagementAllRoot` here?

**Review by Imran92 — APPROVED** [review]
LGTM, just left a small comment

File: packages/domains-table/src/utils/paths.ts (hunk 1 above)
Comment (bogiii) [reply packages/domains-table/src/utils/paths.ts]: True, I'll update it.

**Pull request merged**